src/baldr/nodeinfo.cc

```diff
@@ -221,10 +221,15 @@ void NodeInfo::set_stop_index(const uint32_t stop_index) {
 // Get the heading of the local edge given its local index. Supports
 // up to 8 local edges. Headings are expanded from 8 bits.
 uint32_t NodeInfo::heading(const uint32_t localidx) const {
-  // Make sure everything is 64 bit!
-  uint64_t shift = localidx * 8; // 8 bits per index
-  return static_cast<uint32_t>(std::round(
-      ((headings_ & (static_cast<uint64_t>(255) << shift)) >> shift) * kHeadingExpandFactor));
+  if (localidx > kMaxLocalEdgeIndex) {
```

Comment:
can this actually happen? this should be limited by `nodeinfo::local_edge_count_` which is 3 bits

Reply:
It does happen in the unit tests in my experience and others also ran into this issue (I don't know how exactly). See the discussion from the [initial PR](https://github.com/SvetlanaBayarovich/valhalla/pull/1#discussion_r459616364)

Reply:
@kevinkreiser https://github.com/valhalla/valhalla/issues/2494

Reply:
The point is it shouldn't happen and indicates another bug upstream, right? Instead of papering over it we should follow the call stack and figure our why bogus values are being sent here and fix that

Reply:
Oh, I assumed it might be an expected behavior as long as there was such check in `set_heading`.  I'll try to look into it considering your remark regarding `local_edge_count_`

Reply:
@kevinkreiser yes.  I was questioning the same on the initial PR.  @dnesbitt61 any idea on why we would send in high values here or is it something we need to dig into

Reply:
May need to dig into this more. As I have mentioned, NodeInfo::heading is a field that could/should be deprecated in the future. Its only use outside of data creation is within map matching to find a "turn cost" based on heading changes. This should really use some form of the costing object or use the directed edge turn_type information. In meili the heading method is protected from using a local_index > 7. 
I think the other uses of heading (GraphEnhancer and ShortcutBuilder) could use local storage of headings rather than actually storing them in the tile. Here is where some high values perhaps could be sent in - I haven't tracked this down recently.

Reply:
The first time I encounter this issue is during ` "Building Utrecht Tiles..."` custom command, namely on 
```
  COMMAND ${CMAKE_BINARY_DIR}/valhalla_build_tiles
      --inline-config '{"mjolnir":{"id_table_size":1000,"tile_dir":"test/data/utrecht_tiles","timezone":"test/data/tz.sqlite","admin":"${VALHALLA_SOURCE_DIR}/test/data/netherlands_admin.sqlite","hierarchy":true,"shortcuts":true,"concurrency":1,"logging":{"type":""}}}'
      -s build -e cleanup
      ${VALHALLA_SOURCE_DIR}/test/data/utrecht_netherlands.osm.pbf
```
The callstack at that moment is
```
...
10 valhalla::baldr::NodeInfo::heading(unsigned int) const      nodeinfo.cc   227
11 (anonymous namespace)::GetTurnTypes()                       graphenhancer.cc   214    
12 (anonymous namespace)::UpdateTurnLanes()                    graphenhancer.cc    418  
13 (anonymous namespace)::enhance(...)                         graphenhancer.cc    1790    
...
```
When during second pass node with index `1116` is processed, it has `edge_index` equal to `3310` (which is a valid value for edge?) and without any further checks it goes down to `GetTurnTypes` function which requests heading

Reply:
I see that the index to get heading is calculated as `nodeinfo.edge_index() + j` where `j` is `j = 0; j < nodeinfo.edge_count(); j++`. So this does not look like a correct calculation for `localidx` to be used in this function. The `heading` value is needed for `GetTurnTypes` function. If enhancement should be performed for each edge (not only for those 8 local edges) need to get heading from somewhere else

Reply:
Regarding this PR, should I create a separate issue for this problem and drop this change from in order to deal with it separately?
What's your opinion @kevinkreiser @dnesbitt61 ?

Reply:
i think that would be great yes. it think its clear that something is wrong with the `UpdateTurnLanes` code. It assumes that certain objects are gauranteed to be in vectors but i think sometimes the vectors are empty and so it uses junk data. splitting that out and working on that separately is a good idea

Reply:
Dropping this change. Separate issue was created